c/check.py

```diff
@@ -279,8 +368,40 @@ def check_file_has_no_windows_line_ending(self):
         if any('\r' in line for line in self.lines):
             self.error("Windows line endings")
 
+    def check_no_unknown_property(self):
+        # Check each file name part immediately after a _true- or _false- for a valid property
+        for part in (self.filename.split("_true-")[1:] + self.filename.split("_true-")[1:]):
+            if not any(part.startswith(prop) for prop in PROPERTIES):
+                self.error("has unknown property " + part)
+
+    def check_no_duplicate_verdicts(self):
+        for prop in PROPERTIES:
+            if self.filename.count(prop) > 1:
+                self.error("has duplicate verdict for property " + prop)
+
+    def check_no_multiple_memsafety_verdicts(self):
+        # SV-COMP rules say at most one of these may appear
+        if (self.filename.count("valid-deref") + self.filename.count("valid-free") +
+                self.filename.count("valid-memtrack") + self.filename.count("valid-memsafety")) > 1:
+            self.error("has verdicts for multiple memsafety properties")
+
+    def check_no_contradicting_verdicts(self):
+        def violates(prop):
+            return ("_false-" + prop) in self.filename
+
+        if (violates("valid-deref") or violates("valid-free") or violates("no-overflow") or
+                violates("def-behavior")):
+            if ("_true-" in self.filename) or self.filename.count("_false-") > 1:
+                self.error(
+                    "has expected undefined behavior but also a verdict for some other property")
+
+        if violates("unreach-call") and "_true-valid-memsafety" in self.filename:
+            # __VERIFIER_error() aborts the program, and it is not clear whether allocated memory
+            # at this point would count as memory leak, so we forbid this combination
```

Comment:
Maybe too strict? For simple programs without allocations (ntdrivers-simplified?) this might be a valid combination of properties.